File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function(s) {
    let lvl = 0;
    let res = '';
    for(let i=0;i<s.length;i++) {
        if(s[i]=='(') {
            if(lvl>0) res+=s[i];
            lvl++;
        }else {
            lvl--;
             if(lvl>0) res+=s[i];
        }
    } return res;
    
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {

        String res = "";
        int c = 0;

        for(int i=0;i<s.length();i++){

            char curr = s.charAt(i);

            if(s.charAt(i)=='(') {

                if(c>0) res+=curr;
                c++;

            }else {
                c--;
                if(c>0) res+=curr;
            }
        }
        return res;
        
    }
}